File: java/src/main/java/io/vedadb/wire/vbp/VBPException.java
package io.vedadb.wire.vbp;

/** Wraps a sqlstate for callers that want typed exception access. */
public class VBPException extends RuntimeException {
    public final String sqlstate;
    public VBPException(String sqlstate, String message) {
        super("[" + sqlstate + "] " + message);
        this.sqlstate = sqlstate;
    }
}

File: php/src/VedaDB/Wire/Vbp/VBPException.php
<?php
declare(strict_types=1);

namespace VedaDB\Wire\Vbp;

/**
 * High-level VBP exception (decoded from an ERROR frame body).
 *
 * Mirrors the Java VBPException. Carries a SQLSTATE so callers can branch
 * on the cause without parsing the message.
 */
class VBPException extends \RuntimeException
{
    public function __construct(
        public readonly string $sqlstate = '0A000',
        string $message = '',
        public readonly string $detail = '',
        public readonly string $hint = '',
        ?\Throwable $previous = null,
    ) {
        parent::__construct(
            $message !== '' ? "[$sqlstate] $message" : "[$sqlstate]",
            0,
            $previous,
        );
    }
}
